### ISSUE TITLE
Solaris software RAID via metastat didn't work

### ISSUE THREAD

**Issue by rfertig** [glensc/nagios-plugin-check_raid#38]
It is said, that Solaris software RAID via metastat is supported, but it seems to me, that it didn't work:

```
# /usr/local/nagios/libexec/check_raid.pl
No RAID configuration found (tried: lsscsi, mdstat, megacli, ips, aaccli, afacli, mpt, gdth, dpt_i2o, tw_cli, arcconf, megarc, cmdtool2, cciss, sas2ircu, hpacucli, areca)
# metastat -c
d60              m  5.9GB d61 d62
    d61          s  5.9GB c1t1d0s6
    d62          s  5.9GB c1t0d0s6
d30              m   23GB d31 d32
    d31          s   23GB c1t1d0s3
    d32          s   23GB c1t0d0s3
...
```

OS is Solaris 10:

```
# cat /etc/release
                      Solaris 10 10/09 s10s_u8wos_08a SPARC
           Copyright 2009 Sun Microsystems, Inc.  All Rights Reserved.
                        Use is subject to license terms.
                           Assembled 16 September 2009
```

**Comment by noblemtw** [comment]
I've just committed test data for the metastat plugin, although as it's the first time I've used github I'm not entirely sure if I need to do anything else or how to link my branch to this issue.
In addition to adding the test data, the original code didn't seem to work quite right, but just needed a fix to the regex. It should also now auto-detect whether Solaris Volume Manager (SVM aka SDS or disksuite) is actually being used for software RAID and disable the plugin if it's not configured. Solaris 10 comes pre-installed with metastat so the command is always present, but will error if SVM isn't in use, so this saves having do manually list the plugins that need to be enabled

I've also got additional modules for ZFS software raid as well as LSI based H/W raid via the raidctl command, both of which are used on Solaris, however I'm unsure how to contribute these
